clevertap-core/src/main/java/com/clevertap/android/sdk/CTXtensions.kt

```diff
@@ -143,3 +145,64 @@ fun CleverTapAPI.flushPushImpressionsOnPostAsyncSafely(logTag: String, caller: S
         e.printStackTrace()
     }
 }
+
+fun JSONArray?.isInvalidIndex(index: Int): Boolean {
+    return this == null || index < 0 || index >= this.length()
+}
+
+/**
+ * Extension function to check if a SharedPreferences file has data.
+ * @return `true` if the SharedPreferences file has data, `false` otherwise.
+ */
+fun SharedPreferences.hasData(): Boolean {
+    return all.isNotEmpty()
+}
+
+fun JSONArray?.orEmptyArray(): JSONArray {
+    return this ?: JSONArray()
+}
+
+inline fun <reified T> JSONArray.toList(): List<T> {
+    val list = mutableListOf<T>()
+    for (index in 0 until length()) {
+        val element = get(index)
+        if (element is T) {
+            list.add(element)
+        }
+    }
+    return list
+}
+
+inline fun <reified T> JSONArray.iterator(foreach: (element: T) -> Unit) {
```

Comment:
JsonArray iterator objects are zero/multiple JsonObject, why do we need to make this generic specific?

Reply:
@CTLalit here are the use cases.

1. https://github.com/CleverTap/clevertap-android-sdk/blob/client_side_inapps/clevertap-core/src/test/java/com/clevertap/android/sdk/CTXtensionsTest.kt#L551
2. https://github.com/CleverTap/clevertap-android-sdk/blob/client_side_inapps/clevertap-core/src/test/java/com/clevertap/android/sdk/CTXtensionsTest.kt#L566
3. https://github.com/CleverTap/clevertap-android-sdk/blob/client_side_inapps/clevertap-core/src/test/java/com/clevertap/android/sdk/CTXtensionsTest.kt#L581
4. https://github.com/CleverTap/clevertap-android-sdk/blob/client_side_inapps/clevertap-core/src/test/java/com/clevertap/android/sdk/CTXtensionsTest.kt#L599
5. https://github.com/CleverTap/clevertap-android-sdk/blob/client_side_inapps/clevertap-core/src/test/java/com/clevertap/android/sdk/CTXtensionsTest.kt#L624

Reply:
We made it generic just to use it in test cases?